UGENE-7365. Move columns from MSA to a new document

yalgaer opened the pull request:
To move rows easily between alignment objects we need an easy way to create a new document, which plays the role of exclude list. This change adds a menu option to move sequences to a new file.
https://ugene.dev/tracker/browse/UGENE-7365

I also had to move export task from the export plugin to corelibs (that was an old TODO), document the changed code and fix warnings. The GUI test will be added tomorrow, first I want to see this changed does not break anything.

yalgaer committed:
Initial implementation of moving selected MSA rows to a new document

yalgaer committed:
Add move action to the menu

yalgaer committed:
Remove duplicated files

yalgaer committed:
Merge remote-tracking branch 'origin/master' into UGENE-6375

yalgaer committed:
Add Task suffix to the export task

yalgaer committed:
Minor code cleanup

yalgaer committed:
Minor code cleanup

DmitriiSukhomlinov commented on src/corelibs/U2Core/src/tasks/ExportTasks.cpp:
```diff
@@ -223,31 +189,25 @@ void ExportMSA2MSATask::run() {
-        return;
-    }
+    Document *d = chromaObject->getDocument();
+    SAFE_POINT_EXT(d != nullptr, stateInfo.setError("Chromatogram object document is not found!"), );
```
`L10N::nullPointerError` could be used for this type of errors

yalgaer replied:
Done

DmitriiSukhomlinov commented on src/corelibs/U2Core/src/tasks/ExportTasks.cpp:
```diff
@@ -223,31 +189,25 @@ void ExportMSA2MSATask::run() {
     GObject *resObj = d->findGObjectByName(seqObjName);
     U2SequenceObject *sObj = qobject_cast<U2SequenceObject *>(resObj);
-    assert(sObj != nullptr);
+    SAFE_POINT_EXT(sObj != nullptr, stateInfo.setError(L10N::internalError()), );
```
Maybe `L10N::nullPointerError`

yalgaer replied:
Done

yalgaer committed:
Add some comments

yalgaer committed:
Code review fixes

yalgaer committed:
Fix typo

DmitriiSukhomlinov commented on src/corelibs/U2View/src/ov_msa/move_to_object/MoveToObjectMaController.cpp:
```diff
@@ -127,4 +129,68 @@ void MoveToObjectMaController::buildMenu(GObjectView *, QMenu *menu, const QStri
+/* RemoveRowsFromMsaObject */
+/************************************************************************/
+RemoveRowsFromMaObjectTask::RemoveRowsFromMaObjectTask(MaEditor *_maEditor, const QList<qint64> &_rowIds)
+    : Task(tr("Remove rows from alignment"), TaskFlag_RunInMainThread), maEditor(_maEditor), rowIds(_rowIds) {
```
Why do you need to run it in the main thread? The database updating (`U2UseCommonUserModStep`) could be a bit time-consuming

yalgaer replied:
This task performs UI ops.
Is there any place in our code where U2UseCommonUserModStep are run in non-main thread?

DmitriiSukhomlinov reviewed (changes requested):
Do you want to add several counters to this feature?

yalgaer committed:
Add counters to MoveToMaObject code

Fix GUITest_common_scenarios_msa_editor_test_0056

Add object menu icon to the Move MSA to object action

Fix for GUITest_common_scenarios_project_sequence_exporting_test_0001

Fix for GUITest_regression_scenarios_test_3656

Fix for GUITest_regression_scenarios_test_3402

Add GUITest_common_scenarios_msa_editor_test_0096

yalgaer committed:
Merge remote-tracking branch 'origin/master' into UGENE-6375

yalgaer commented:
> Do you want to add several counters to this feature?

Added
